engine/src/main/java/org/terasology/rendering/cameras/Camera.java

```diff
@@ -204,6 +206,22 @@ public Vector3f getViewingDirection() {
         return viewingDirection;
     }
 
+    /**
+     * Get the gaze direction.
+     * @return the gaze direction, a quaternion.
+     */
+    public Quat4f getGazeDirection() {
```

Comment:
So in this case, where the `LocalPlayerSystem` manages player movement, it is necessary to have some way of checking whether the orientation needs to be updated by events or whether it is being set by a tracking system. I have named this method `isTracked()`. To respond to your desire to have less work/logic happening in the camera (as it is moving toward being a `Component`), I have moved the logic relating to how the camera movement is handled into `LocalPlayerSystem`.

Moving forward I think there will be a `RoomscaleVRPlayerSystem` and `SeatedVRPlayerSystem` to take the burden of making this decision away from this VR-nonspecific system. In that case it won't be necessary to have the check on the camera, because we will already know that a VR camera is active.